### PR TITLE
Refactor run to reuse eval functions

### PR DESCRIPTION
**Pull request by anneouyang**

### CODE REVIEW

File: src/prompt_constructor.py [hunk 1]
```diff
@@ -52,11 +52,12 @@ def prompt_generate_custom_cuda(arc_src: str,
     """
     return prompt
 
-def prompt_generate_custom_cuda_from_file(arch_path, example_ind=0):
+def prompt_generate_custom_cuda_from_file(ref_arch_src, example_ind=0):
```
Comment (simonguozirui) [inline src/prompt_constructor.py:55]: curious why the change here? the dataset points to arch_path, wouldn't it be easier to read from path directly?

Comment (simonguozirui) [reply src/prompt_constructor.py:55]: is it to make it easier for multi-turn?

**Comment by simonguozirui** [comment]
Thanks! Let's keep the eval/helper/utils logic between various codebases the same

File: src/prompt_constructor.py (hunk 1 above)
Comment (anneouyang) [reply src/prompt_constructor.py:55]: because im trying to reuse the `fetch_ref_arch_from_problem_id` and `construct_problem_dataset_from_problem_dir` functions
